frontend/src/pages/RunDetails.test.tsx

```diff
@@ -167,6 +168,9 @@ describe('RunDetails', () => {
     // Hide expected warning messages
     warnSpy = jest.spyOn(Utils.logger, 'warn').mockImplementation();
 
+    // Mock decodeCompressedNodes to avoid dequeuing chained promises
+    decodeCompressedNodesSpy = jest.spyOn(Utils, 'decodeCompressedNodes').mockImplementation();
```

Comment:
nit: `decodeCompressedNodes` itself is very important to test in this test case, because otherwise we are not verifying this feature end to end.

I understand you may be blocked by sth. You can put a TODO comment here too if you cannot figure out by yourself.

If you'd want to try testing it. Here are my ideas:
I'm not sure how `zlib` will trigger the callback, I guess if simply await is not enough, we can use `setTimeout` to explicitly wait some long time period so `zlib` can decompress.

Reply:
Well, I already unit tested `decodeCompressedNodes` in  [Utils.test.tsx](https://github.com/kubeflow/pipelines/pull/4180/files#diff-3c4323c8f2a90f3a3b926a319d85baf1R267). 
The `setTimeout` trick will work for sure but not so elegant.

Reply:
I test the wait and it works, thanks for the hint: 8491c5c6104fcaee170d1b0b6246d654dd7a25b8